mypy/checkmember.py

```diff
@@ -630,7 +630,7 @@ class B(A): pass
 
     """
     if isinstance(method, Overloaded):
-        return cast(F, Overloaded([bind_self(c, method) for c in method.items()]))
+        return cast(F, Overloaded([bind_self(c, original_type) for c in method.items()]))
```

Comment:
Wow, was that a bug?

Reply:
Indeed. I recall fixing it in at some point, but probably only on a local branch.

The reason it hadn't surface (AFAIK) is that it requires an overloaded method with generic self that's also has as bound a supertype of the class, without overlapping the other overload. This ish is a rare combination which is hard to get right anyway, and the checking for the "being a supertype" part is too strict in the case of overload, so I actually struggle to make an example where this is the _only_ error.

```python
from typing import TypeVar, overload

T = TypeVar('T')

class A:
    @overload
    def f(self: T) -> T: return self
    def f(self): return self

reveal_type(A().f())
```
result:
```
../tmp.py:6: error: Single overload definition, multiple required
../tmp.py:10: error: Revealed type is 'Overload(def [T] (self: T`9) -> T`9)'
```
Adding a valid overload is not obvious (to me).